rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYamlVisitor.java

```diff
@@ -89,96 +100,160 @@ public Yaml visitSequence(Yaml.Sequence existingSeq, P p) {
     }
 
     @Override
-    public Yaml visitMapping(Yaml.Mapping existingMapping, P p) {
-        if (scope.isScope(existingMapping) && incoming instanceof Yaml.Mapping) {
-            return mergeMapping(existingMapping, (Yaml.Mapping) incoming, p, getCursor());
+    public Yaml visitMapping(Mapping existingMapping, P p) {
+        if (existing.isScope(existingMapping) && incoming instanceof Mapping) {
+            Mapping mapping = mergeMapping(existingMapping, (Mapping) incoming, p, getCursor());
+
+            if (getCursor().getMessage(REMOVE_PREFIX, false)) {
+                List<Entry> entries = ((Mapping) getCursor().getValue()).getEntries();
+                return mapping.withEntries(mapLast(mapping.getEntries(), it -> it.withPrefix("\n" + grabPartLineBreak(entries.get(entries.size() - 1), 1))));
```

Comment:
Doing `entries.get(entries.size() - 1),` over and over again is not my cup of tea. Maybe we should consider a `last` function in the ListUtils class...